client/src/util/stateManager/matrix.ts

```diff
@@ -12,30 +12,68 @@ import {
   KeyIndex,
 } from "../dataframe";
 
+import { CatInt8Array } from "./cat8_array";
+import { CatInt16Array } from "./cat16_array";
+import { CatInt32Array } from "./cat32_array";
+
 const utf8Decoder = new TextDecoder("utf-8");
 
 /**
  * Matrix flatbuffer decoding support. See fbs/matrix.fbs
  */
 
 /**
- * Decode NetEncoding.TypedArray
+ * Decode NetEncoding.TypedFBArray
  */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-function decodeTypedArray(uType: any, uValF: any, inplace = false) {
-  if (uType === NetEncoding.TypedArray.NONE) {
+function decodeTypedArray(uType: any, uValF: any, inplace = false): any {
+  if (uType === NetEncoding.TypedFBArray.NONE) {
     return null;
   }
 
   // Convert to a JS class that supports this type
   // @ts-expect-error --- FIXME: Element implicitly has an 'any' type.
-  const TypeClass = NetEncoding[NetEncoding.TypedArray[uType]];
+  const TypeClass = NetEncoding[NetEncoding.TypedFBArray[uType]];
   // Create a TypedArray that references the underlying buffer
-  let arr = uValF(new TypeClass()).dataArray();
-  if (uType === NetEncoding.TypedArray.JSONEncodedArray) {
+  const x = uValF(new TypeClass());
+  let arr = x.dataArray();
+  let dict;
+  let flag = true;
+  if (uType === NetEncoding.TypedFBArray.JSONEncodedFBArray) {
```

Comment:
this whole section (if/else/else...) would be clearer if it was refactored into a set of stand-alone functions which decode a single type, eg,  `decodeJSONTypedArray()`, `decodeSparseTypedArray()`, etc.